File: clients/r/tests/integration.R
Sys.setenv(LC_ALL="C")
source("../R/orbit_api_client.R")

if (Sys.getenv("ORBIT_INTEGRATION") != "1") {
  cat("Skipping integration test. Set ORBIT_INTEGRATION=1 to enable.\n")
  quit(status=0)
}

url <- Sys.getenv("ORBIT_URL", "http://localhost:3000")
client <- orbit_api_client(url)
buf <- ""
stream_chat(client, "ping", FALSE, function(chunk) { buf <<- paste0(buf, chunk$text) })
if (nchar(buf) == 0) { stop("Empty response") } else { cat("OK\n") }

File: clients/php/tests/integration.php
<?php
require_once __DIR__ . '/../src/ApiClient.php';
use SchmiTech\Orbit\ApiClient;

if (getenv('ORBIT_INTEGRATION') !== '1') {
  fwrite(STDERR, "Skipping integration test. Set ORBIT_INTEGRATION=1 to run.\n");
  exit(0);
}

$url = getenv('ORBIT_URL') ?: 'http://localhost:3000';
$client = new ApiClient($url);
$all = '';
$client->streamChat('ping', false, function($chunk) use (&$all) { $all .= $chunk->text; });
if ($all === '') { fwrite(STDERR, "Empty response\n"); exit(1); }
echo "OK\n";
exit(0);

